### DISCUSSION TITLE
Scenario list and OpenScenarios Idea

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#19134] [Feature requests]
1:1 Substitutions are desired for OpenScenarios, and the community should try making them...
Discussing with the devs, the intent is 1 substitution per scenario in the base game.
so OpenScenarios 1.0 release would have a substitute for each RCT2 Vanilla (non expansion) scenario, and then be incorporated into the actual main release. Later versions would include expansion scenario and rct1 scenario substitutions.

Here is an idea I have for OpenScenarios:

Cooky Castle
Electric Meadow
Industrial Capers
Altruist Airfield
Bustling Botany
Bustling Bazaar
Musty Greens
Fungus Forest
Inertia Gardens
Infernal Venues 
Azimuth Adventures
Extreme Hills
Haunted Town
Lucky River
Rainbow Heights

I also want to know if the community thinks it would be better to:
1. ) Recreate all scenarios to be almost the same as the original
2. ) Substitute all scenarios with completely different scenarios (15)
3. ) Remix the scenarios: for example changing the goals, difficulty, or theming (example: grass -> ice), keeping the difficulty curve, but maybe making the scenarios that were difficult easy and the ones that were easy difficult.

You can find out more about OpenScenarios in the OpenScenarios github page.

**Comment by ocalhoun6** [comment]
>) Recreate all scenarios to be almost the same as the original

If you're doing that, then what's the point of replacing them?

>) Replace all scenarios with completely different scenarios (15)

I'd be fine with *adding* new scenarios, but I wouldn't want to *replace* the originals -- ie, I wouldn't want to lose any of the original scenarios.

I do think it would be nice to have an official OpenRCT2 scenario group -- a set of impeccably made scenarios that ships with OpenRCT2 as a standardized group unique to this version of the game. (Maybe it could make good use of the new save format's increased limits, and/or make use of entirely new types of scenario goals.)

>) Remix the scenarios: for example changing the goals, difficulty, or theming (example: grass -> ice), keeping the difficulty curve, but maybe making the scenarios that were difficult easy and the ones that were easy difficult.

Along those lines, I think the best way to do things would be to develop a way to produce semi-random, procedurally generated scenarios. (I think there's already a plugin for this, but it would be great to build it into OpenRCT2 itself, giving a 'RANDOM' option in the scenario menu when starting a new game.)

Basically, you click the 'randomized scenario' button and the game automatically:

1: Uses the terrain generator that already exists in the scenario editor + a few randomized settings within reasonable parameters to build the basic landscape.

2: Land ownership: Generates a square-ish shape (randomized with a bit of noise to produce rectangles and the occasional odd shape, but always contiguous) and applies that shape to the map, setting that area as owned land. Then has a random chance (say 50%) of setting an additional area bordering that area as land for sale. Then has a smaller chance (say 10%) of setting an additional bordering area as either construction rights owned or construction rights for sale.

3: (Possibly the most difficult part to code) Paths and entrances: places a random number of park entrances (up to 4) on the park borders established in step 2. Automatically builds paths between those entrances and the edge of the map, then places peep spawn points where the path meets the edge of the map. (If any of those paths happen to be in for sale or construction rights zones, sets those individual tiles with path as not owned.)

4: Sets the invention list and available objects semi-randomly. *Mostly* random selection, but always guaranteed to have a basic set of stalls and at least 1 of each ride type already invented.

5: Sets the scenario goals and parameters randomly, within certain hard-coded reasonable limits. Including selecting a random goal, semi-random amount of money/loan, and a random chance of selecting any of the special parameters like more difficult guest generation or no advertising. Could also have a random chance of selecting a 'no money' scenario *if* the scenario goal is one of the ones compatible with that.

6: Saves the randomly generated scenario in a different directory (*not* the saved game directory, nor the scenarios directory). This will allow players an option to play the same scenario over again from the beginning, which they might want to do if they lost the first time and want to try it again. It would also allow the player to share the randomly generated scenario with others if it happened to be really good. Including setting the park/scenario name, by combining words from a pre-selected list like 'adjective noun' or 'noun world', etc.

7: Opens the scenario and starts a new game with it.

**Reply by 733737** [reply]
When I said replace I meant substitute for when openrct2 reaches 1.0 and is similar to openttd in that it is a standalone game

**Comment by HtotheTML** [comment]
A lot of the base game RCT2 maps suffer from being well...empty. In addition, the terraforming is largely uninspired. It would be more than enough to give the scenarios a glow-up to meet higher quality standards and update their ride selection to include rides introduced in Open.

So a remix would be the best way to go, I think.

**Comment by HtotheTML** [comment]
Adding more rides and scaling the difficulty of the objectives better would also be helpful, even perhaps changing a few objectives slightly.

Even with the recent reordering of the scenarios, the difficulty curve is still rather wonky. Your best bet with difficulty curves is to play 1's scenarios before playing 2's.

**Comment by KatieZeldaKat** [comment]
Given the variety in people's preferences for scenarios, I think what would truly be best is to allow players to customize the scenarios in the list themselves and make tweaks/additions that way. Having custom tabs in the scenarios list with custom parks (each categorized as Beginner, Challenging, or Expert, ideally in a custom order that's not just alphabetical) would greatly increase players abilities to not only have their own improved scenarios, but also let scenario packs be imported as a packed bundle in a dedicated tab. This wouldn't require any compromising on the OpenRCT2's team to dictate which scenarios would make the final cut in the "actual main release," as you say.

This import process could also ensure people could use the new .park files to replace the current RCT1 and RCT2 scenarios, as the only way to replace these scenarios currently is to ensure they are an .sc4, .sc6, or .sea, all limited formats which require other programs or older versions of OpenRCT2 to manipulate. Personally, I wouldn't know what the best way of replacing these scenarios would be, or even if it would be necessary. If people could create their own pack and perhaps name it "RCT1 Remixed", then there might not be a need to replace the scenarios at all in the end! People would just play the packs they feel embody their vision of the game and their game style the best. Allowing players to hide certain tabs would de-clutter the interface though.

**Comment by MASHzj** [comment]
There is no need to literally "replace" original scenarios by new ones. Following the discussion about open scenarios on Discord, it seems the intention to make a separate tab with 30 new parks, probably called "OpenRCT2" or "Open scenarios". The other tabs would still stay (RCT1, rct2 and their expansions) next to it. I'm looking forward to which scenarios will eventually make it into this list. But I still would want to play original rct1 and rct2 scenarios too, mostly out of nostalgia, and I guess I'm not the only one.

**Reply by 733737** [reply]
I updated the original post to clarify
